pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/RandomAccessSequenceAdapter.java

```diff
@@ -138,6 +138,9 @@ public long readBytes(@NonNull final ByteBuffer dst) {
         final var length = Math.min(dst.remaining(), remaining());
         final var finalLimit = dstPos + Math.min(length, dst.remaining());
         dst.limit(Math.toIntExact(finalLimit));
+        if (length == 0) {
```

Comment:
nit: possibly move this test before the costly math operations.

Reply:
Nope, that's the catch. I originally did just that and discovered a lot of tests fail. The tests rely on a side-effect of all these operations that actually call dst.limit() at line 140, so we cannot skip the call and return earlier.

That's a terrible side-effect and an undocumented one. But that's what today's code relies on. So I decided not to break the existing code. We'll need a separate issue if we want to remove this side-effect. Sadly, it's present in a few places, not just this one.

Reply:
Ahh.  That is, indeed, a very unfortunate (and undesirable) condition.
Perhaps file an issue to eventually fix that side effect reliance(so it isn't forgotten)?

Reply:
I filed https://github.com/hashgraph/pbj/issues/194